packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -26,26 +42,91 @@ export class Wallet {
    */
   public static async create(
     engine: Engine,
+    chainService: ChainServiceInterface,
     messageServiceFactory: MessageServiceFactory,
-    retryOptions: Partial<RetryOptions> = DEFAULTS
+    retryOptions: Partial<SyncOptions> = DEFAULTS
   ): Promise<Wallet> {
-    return new Wallet(messageServiceFactory, engine, {...DEFAULTS, ...retryOptions});
+    await chainService.checkChainId(engine.engineConfig.networkConfiguration.chainNetworkID);
+    const wallet = new Wallet(messageServiceFactory, engine, chainService, {
+      ...DEFAULTS,
+      ...retryOptions,
+    });
+    await wallet.registerExistingChannelsWithChainService();
+    return wallet;
   }
 
   private _messageService: MessageServiceInterface;
 
+  private _syncInterval = setIntervalAsync(async () => {
+    await this.syncObjectives();
+  }, this._syncOptions.pollInterval);
+
+  private _chainListener: ChainEventSubscriberInterface;
   private constructor(
     messageServiceFactory: MessageServiceFactory,
     private _engine: Engine,
-    private _retryOptions: RetryOptions
+    private _chainService: ChainServiceInterface,
+    private _syncOptions: SyncOptions
   ) {
+    super();
+
     const handler: MessageHandler = async message => {
-      const {outbox} = await this._engine.pushMessage(message.data);
+      const result = await this._engine.pushMessage(message.data);
+      const {channelResults} = result;
+
+      await this.registerChannels(channelResults);
 
-      await this.messageService.send(getMessages(outbox));
+      await this.handleEngineOutput(result);
     };
 
     this._messageService = messageServiceFactory(handler);
+
+    this._chainListener = {
+      holdingUpdated: this.createChainEventlistener('holdingUpdated', e =>
+        this._engine.store.updateFunding(e.channelId, e.amount, e.assetHolderAddress)
+      ),
+      assetOutcomeUpdated: this.createChainEventlistener('assetOutcomeUpdated', async e => {
+        const transferredOut = e.externalPayouts.map(ai => ({
+          toAddress: makeDestination(ai.destination),
```

Comment:
Why is this keyed `toAddress` and not `destination`?

(and why is `ai.destination` not a `Destination`?)

Reply:
This was copied from the original behaviour in the `engine` so I'm not sure why those choices were made.

Reply:
https://github.com/statechannels/statechannels/issues/3594